src/providers/ethereumProviderWrapper.ts

```diff
@@ -125,21 +130,65 @@ export class EthereumProviderWrapper {
     }
   }
 
-  async getAllData(address: string, keys: string[]) {
+  async getAllData(address: string, keys: string[]): Promise<GetDataReturn[]> {
+    const erc725Version = await this.getErc725YVersion(address);
+
+    if (erc725Version === ERC725_VERSION.NOT_ERC725) {
+      throw new Error(
+        `Contract: ${address} does not support ERC725Y interface.`,
+      );
+    }
+
+    switch (erc725Version) {
+      case ERC725_VERSION.ERC725:
+        return this.getAllDataNonLegacy(address, keys);
+      case ERC725_VERSION.ERC725_LEGACY:
+        return this.getAllDataLegacy(address, keys);
+      default:
+        return [];
+    }
+  }
+
+  private async getAllDataNonLegacy(
```

Comment:
what a horrible name :)

`getData` should be it and if there is OLD function you call it `getDataLegacy`

Reply:
There is already a function named getData, it was the one used in previous code as there was no distinction between legacy and no legacy. I didn't know what else I could use as getData was already taken.... If I call this one getData, how do I call the other one then?

Reply:
I'm thinking maybe the current "getData" function can be removed at least for new ERC725Y as the command is the same, only the number of items in the array is different. I will check this tomorrow and make it more "lean". There might be some useless logic duplication